apps/src/lib/util/firehose.js

```diff
@@ -109,14 +109,63 @@ class FirehoseClient {
   }
 
   /**
+   * Returns a unique user ID that is persisted across sessions through local
+   * storage.
+   * WARNING: Mutates local storage if an analytisID has not already been set.
```

Comment:
Something we may want to consider here, too, if anyone needs to look at longer term metrics - if the user is signed in to CDO, we might want to do something special here, since this ID will stick to the browser, and in a lot of schools each student ends up using a different computer every time without wiping cookies/local storage.